src/Acuminator/Acuminator.Analyzers/StaticAnalysis/PXOverrideMismatch/PXOverrideMismatchAnalyzer.cs

```diff
@@ -0,0 +1,166 @@
+		{
+			// Here we know that the method is not static and has the correct attribute.
+
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			foreach (var baseType in allBaseTypes)
+			{
+				var allMembers = baseType.GetMembers(methodSymbol.Name);
+
+				if (!allMembers.IsDefaultOrEmpty)
+				{
+					if (allMembers.OfType<IMethodSymbol>().Any(m => PXOverrideHelper.IsSuitable(methodSymbol, m)))
+					{
+						return;
+					}
+				}
+			}
+
+			var location = methodSymbol.Locations.FirstOrDefault();
+
+			if (location != null)
+			{
+				var diagnostic = Diagnostic.Create(Descriptors.PX1096_PXOverrideMustMatchSignature, location);
+
+				context.ReportDiagnosticWithSuppressionCheck(diagnostic, pxContext.CodeAnalysisSettings);
+			}
+		}
+
+		private class PXOverrideHelper
+		{
+			/// <summary>
+			/// Special signature check between a derived method with the PXOverride attribute and the base method.
+			/// </summary>
+			/// <param name="baseMethod">The method from the derived class, with the PXOverride attribute</param>
+			/// <param name="pxOverrideMethod">The method from the base</param>
+			/// <returns></returns>
+			public static bool IsSuitable(IMethodSymbol baseMethod, IMethodSymbol pxOverrideMethod)
+			{
+				var methodsCompatibility = GetMethodsCompatibility(baseMethod.Parameters.Length, pxOverrideMethod.Parameters.Length);
+
+				if (methodsCompatibility == MethodsCompatibility.NotCompatible)
+				{
+					return false;
+				}
+
+				if (!pxOverrideMethod.CanBeOverriden() || !pxOverrideMethod.IsAccessibleOutsideOfAssembly())
+				{
+					return false;
+				}
+
+				if (methodsCompatibility == MethodsCompatibility.ParametersMatch)
+				{
+					return CheckExactMatch(pxOverrideMethod, baseMethod);
+				}
+
+				if (methodsCompatibility == MethodsCompatibility.ParametersMatchWithDelegate)
+				{
+					if (baseMethod.Parameters[baseMethod.Parameters.Length - 1].Type is not INamedTypeSymbol @delegate || @delegate.TypeKind != TypeKind.Delegate)
+					{
+						return false;
+					}
+
+					return CheckExactMatch(pxOverrideMethod, @delegate.DelegateInvokeMethod);
+				}
+
+				return false;
+			}
+
+			private static bool CheckExactMatch(IMethodSymbol pxOverrideMethod, IMethodSymbol delegateInvokeMethod)
+			{
+				if (!pxOverrideMethod.ReturnType.Equals(delegateInvokeMethod.ReturnType))
+				{
+					return false;
+				}
+
+				if (pxOverrideMethod.Parameters.Length != delegateInvokeMethod.Parameters.Length)
+				{
+					return false;
+				}
+
+				if (!DoParametersMatch(pxOverrideMethod.Parameters, delegateInvokeMethod.Parameters))
+				{
+					return false;
+				}
+
+				return true;
+			}
+
+			private static bool DoParametersMatch(ImmutableArray<IParameterSymbol> sourceParameters, ImmutableArray<IParameterSymbol> targetParameters)
+			{
+				for (var i = 0; i < sourceParameters.Length; i++)
+				{
+					if (i >= targetParameters.Length || !sourceParameters[i].Type.Equals(targetParameters[i].Type))
```

Comment:
the check `i >= targetParameters.Length` is unnecessary at this stage. We should have already filtered all suitable base method candidates that have more parameters then PXOverride method. No need to check it, especially in a cycle:
```suggestion
					if (!sourceParameters[i].Type.Equals(targetParameters[i].Type))
```

Reply:
Removed. Thank you.